chapters/exploitation-techniques/shellcodes/drills/08-challenge-shellcode-after/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# syntax=docker/dockerfile:1.4
+#
+# Model: multi-stage build
+#   • build stage: compiles the shellcode exercise
+#   • runtime stage: runs the exe on port 31346
+
+########## Build ##########
+FROM gcc:12 AS builder
+WORKDIR /src
+
+COPY src/ .
+
+RUN make
+
+########## Runtime ##########
+FROM python:3.9-slim
+WORKDIR /app
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends binutils cpp && \
+    pip install pwntools==4.14.1
+
+COPY --from=builder /src/vuln /app/vuln
+COPY sol/ /app/
+
+EXPOSE 31346
+
+CMD ["/bin/sh", "-c", "/app/vuln & python exploit.py"]
```

Comment:
What is the point of running the exploit here? This should only start the vulnerable app, right?